src/main/java/seedu/pill/command/HelpCommand.java

```diff
@@ -0,0 +1,255 @@
+    }
+
+    /**
+     * Executes the help command by displaying information about available commands.
+     * @param itemMap       - The item list to be manipulated by the command.
+     * @param storage        -
+     * @throws PillException -
+     */
+    @Override
+    public void execute(ItemMap itemMap, Storage storage) throws PillException {
+        assert itemMap != null : "ItemList cannot be null";
+        assert storage != null : "Storage cannot be null";
+        logger.info("Executing HelpCommand");
+
+        if (commandName == null || commandName.isEmpty()) {
+            showGeneralHelp();
+        } else {
+            showSpecificHelp(commandName);
+        }
+    }
+
+    /**
+     * Displays general help information for every command.
+     */
+    private void showGeneralHelp() {
+        logger.info("Showing general help information");
+
+        System.out.println("Available commands:");
+        System.out.println("  help          - Shows this help message");
+        System.out.println("  add           - Adds a new item to the list");
+        System.out.println("  delete        - Deletes an item from the list");
+        System.out.println("  edit          - Edits an item in the list");
+        System.out.println("  list          - Lists all items");
+        System.out.println("  stock-check   - Lists all items that need to be restocked");
+        System.out.println("  exit          - Exits the program");
+        System.out.println("Type 'help <command>' for more information on a specific command.");
+        System.out.println("Type 'help <command> -v' for verbose output with examples.");
+    }
+
+    /**
+     * Calls the appropriate method depending on what the user requests help for.
+     * @param command - optional user input that determines which help information is displayed to the user.
+     */
+    private void showSpecificHelp(String command) {
+        assert command != null : "Command cannot be null";
+        logger.info("Showing specific help for command: " + command);
+
+        switch (command.toLowerCase()) {
+        case "help":
+            showHelpHelp();
+            break;
+        case "add":
+            showAddHelp();
+            break;
+        case "delete":
+            showDeleteHelp();
+            break;
+        case "edit":
+            showEditHelp();
+            break;
+        case "list":
+            showListHelp();
+            break;
+        case "stock-check":
+            showStockCheckHelp();
+            break;
+        case "exit":
+            showExitHelp();
+            break;
+        default:
+            String closestMatch = StringMatcher.findClosestMatch(command, VALID_COMMANDS);
+            if (closestMatch != null) {
+                System.out.println("Did you mean: " + closestMatch + "?");
+                showSpecificHelp(closestMatch);
+            } else {
+                suggestSimilarCommand(command);
+            }
```

Comment:
Maybe this can be changed to a guard clause instead to make the happy path more prominent